tests/dependency_analyzer/expected_results/cyclonedx_apache_maven.dl

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved. */
+/* Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/. */
+
+#include "prelude.dl"
+
+Policy("check-dependencies", component_id, "Check the dependencies of pkg:github.com/apache/maven@3fc399318edef0d5ba593723a24fff64291d6f9b") :-
```

Comment:
Is it worth converting this dependency resolution result to Souffle? I wonder what would happen if we wanted to create another test case of this type. Are we supposed to write this kind of result by hand?

Reply:
Let me rephrase the question.
I think it is better if I start by asking: What do you think is being tested with this test case? If this test case was meant to test the exhaustive list of dependencies, then I'm not sure if converting it to Souffle is a good idea.

Reply:
All of the check dependency policies were converted from JSON files being passed into the `compare_dependencies.py` e2e script. As far as I'm aware, they exist to exhaustively test dependencies. If we want to change that, I think that would be best done in another PR. The same would apply to other possible optimisations/improvements that have been discussed.

Reply:
What I'm checking here is whether we should convert dependency resolution tests to Souffle, or keep the old way with JSON output and compare with the `compare_dependencies.py` script. Since the expected output for dependency results is rather large and tedious to update by hand, my subjective opinion is keeping the JSON output may be still the better option for now, although others may argue otherwise. If using Souffle here provides no obvious upsides compared to the old way, then I think we may keep it as is. I'm not suggesting any further improvement to be done in this PR otherwise.

Reply:
The script I used to create policies from expected dependency results could be cleaned up and made available. That would be another option for reducing tedium.